### PR TITLE
Sleep for 2ms to avoid time-sensitive test failures

### PR DESCRIPTION
**Pull request by LiranCohen**
Experienced this issue and tracked it down:
https://github.com/TBD54566975/dwn-sdk-js/issues/481

This might be some sort of precision bug within Temporal itself, I'm not really sure how various NodeJS systems handle time precision.

But in these cases there simply needs to be a gap in time for the testing logic, increasing the gap by an additional millisecond will make sure these tests are accurate.

### CODE REVIEW

**Review by thehenrytsai — DISMISSED** [review]
🙏 
I wonder if it's worthwhile introducing a common util method like `minimalSleep()`.

**Comment by LiranCohen** [comment]
@thehenrytsai I really like that idea, let me put that in this PR so we can set precedent.

File: tests/utils/test-data-generator.ts [hunk 1]
```diff
@@ -821,4 +821,12 @@ export class TestDataGenerator {
...
     };
   }
+
+  /**
+   * We must sleep for at least 2ms to avoid timestamp collisions during testing.
+   * https://github.com/TBD54566975/dwn-sdk-js/issues/481
+   */
+  public static async minimalSleep(): Promise<void> {
+    await sleep(2);
+  }
```
Comment (thehenrytsai) [inline tests/utils/test-data-generator.ts:831]: Sorry for the final nit:

Let's put this directly in `time.ts`!

File: tests/utils/test-data-generator.ts [hunk 2]
```diff
@@ -821,4 +821,12 @@ export class TestDataGenerator {
       didDocumentMetadata: {}
     };
   }
+
+  /**
+   * We must sleep for at least 2ms to avoid timestamp collisions during testing.
+   * https://github.com/TBD54566975/dwn-sdk-js/issues/481
```
Comment (thehenrytsai) [inline tests/utils/test-data-generator.ts:827]: Love that you provide context and issue number/URL for historical reference!!

**Review by thehenrytsai — APPROVED** [review]
🙏 🙏 🙏

**Pull request merged**